decisions/2020/02/04-05-tapps-remotes-in-oslo.md

```diff
@@ -0,0 +1,143 @@
+# 2020-02-04 and 05
+
+This minute is different than the others in the sense that we did not
+take comprehensive notes during the multiple days of meetings, so this is a
+recollection at best, and erroneous at worst. Either way, it does not
+attempt to be a detailed document.
+
+It does attempt to capture the overall themes and outcomes.
+
+# Retrospective and planning
+
+During the retrospective we touched on what works well, and laid plans
+for the week.
+
+## Remote culture
+
+The problems [we
+discussed](https://github.com/dhis2/notes/blob/master/decisions/2019/11/11-tapps-meeting-remote-errors-etc.md#remote-culture)
+in November are still real, but there is an attitude that we are working
+on them and cultural changes that time, so we don't expect any
+short-term miracles.
+
+@varl still needs to draft the remote principles into a document. Give
+it a name and it is easier to absorb and talk about.
+
+## Testing
+
+We are improving our automated test suites and actively investing in
+that area to support the overall long-term plan from management on
+stability and robustness.
+
+We have an open agenda (as opposed to hidden) that we want to replace
+Zephyr-tests (manually executed) with Cypress (automatically executed)
+to better support the QA team and processes.
+
+## Team work
+
+We have a good vibe going on, and are better at using the right forum
+for communication (Slack vs. dhis2/notes vs. VoIP).
+
+We still fall into the trap where a developer works solo on a large
+project, and we want to pair up more to tackle the large projects in
+[listed in the agenda](https://github.com/dhis2/notes/issues/89).
+
+We have many examples of apps that have been written by a single dev,
+and do not make sense to any other developer. We consider it one of the
+root causes of our maintainability challenges.
+
+# Testing and strategy
+
+Since this was a big topic we set aside a day for it. We went over some
+proposals, either directly or touched on them during discussions:
+
+- [#90](https://github.com/dhis2/notes/issues/90)
+- [#88](https://github.com/dhis2/notes/issues/88)
+- [#85](https://github.com/dhis2/notes/issues/85)
+
+## Unit vs. End-to-End
+
+We decided that not everything needs to be a Cypress test, but
+user-driven behavior should be tested end-to-end.
+
+For more implementation detail oriented tests, we should prefer Jest.
+
+The important take-away is that we have tests -- and Jest vs. Cypress
+is left for consideration of the implementers and reviwers.
```

Comment:
Not sure if we all fully agreed on this but as I recall it, for scenarios where arguments could be made to either test things in Jest or Cypress, there seemed to be a slight preference for Cypress over Jest, because Cypress is "more real" (i.e. real DOM events etc.).

Reply:
What I remember about the talk abour this is that for business logic we should use Jest in favor of other test frameworks such as hapijs/lab, mocha that we have in for example the App Hub project. Cypress is still our primary choice for testing ui/flows.

Reply:
ae79fa1